### PR TITLE
feat: add Bun to tools

### PR DESCRIPTION
**Pull request by dirien**
Signed-off-by: Engin Diri <engin.diri@mail.schwarz>

## Description

Add Bun to the list of available tools

## How Has This Been Tested?

```bash
❯ docker container run --rm -v $(pwd):/workspace -ti golang:1.17 bash
root@8946fe8f00ed:/go# cd /workspace/
root@8946fe8f00ed:/workspace# go build . 
root@8946fe8f00ed:/workspace# ./arkade get bun
Downloading: bun
2022/08/12 08:21:25 Looking up version for bun
2022/08/12 08:21:26 Found: bun-v0.1.8
Downloading: https://github.com/oven-sh/bun/releases/download/bun-v0.1.8/bun-linux-x64.zip
31.66 MiB / 31.66 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
/tmp/bun-linux-x64.zip written.
2022/08/12 08:21:28 Looking up version for bun
2022/08/12 08:21:28 Found: bun-v0.1.8
Name: bun-linux-x64.zip, size: 331990502022/08/12 08:21:29 Extracted: /tmp/bun
2022/08/12 08:21:29 Copying /tmp/bun to /root/.arkade/bin/bun

Wrote: /root/.arkade/bin/bun (87.83MB)

# Add arkade binary directory to your PATH variable
export PATH=$PATH:$HOME/.arkade/bin/

# Test the binary:
/root/.arkade/bin/bun

# Or install with:
sudo mv /root/.arkade/bin/bun /usr/local/bin/

🐳 arkade needs your support, learn more: https://my.arkade.dev
root@8946fe8f00ed:/workspace# mv /root/.arkade/bin/bun /usr/local/bin/
root@8946fe8f00ed:/workspace# bun
bun: a fast bundler, transpiler, JavaScript Runtime and package manager for web software.

  dev       ./a.ts ./b.jsx        Start a bun Dev Server
  bun       ./a.ts ./b.jsx        Bundle dependencies of input files into a .bun

  init                            Start an empty Bun project from a blank template
  create    next ./app            Create a new project from a template (bun c)
  run       test                  Run JavaScript with bun, a package.json script, or a bin
  install                         Install dependencies for a package.json (bun i)
  add       react                 Add a dependency to package.json (bun a)
  link                            Link an npm package globally
  remove    backbone              Remove a dependency from package.json (bun rm)
  unlink                          Globally unlink an npm package

  upgrade                         Get the latest version of bun
  completions                     Install shell completions for tab-completion
  discord                         Open bun's Discord server
  help                            Print this help menu
```

## Are you a GitHub Sponsor yet (Yes/No?)

- [ ] Yes
- [x] No

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [x] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: pkg/get/tools.go [hunk 1]
```diff
@@ -2571,5 +2571,29 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Name}}
...
+			Description: "Bun is an incredibly fast JavaScript runtime, bundler, transpiler and package manager – all in one.",
+			BinaryTemplate: `
+							{{$os := ""}}
+							{{ if HasPrefix .OS "ming" -}}
+							{{$os = "windows"}}
+							{{- else if eq .OS "linux" -}}
+							{{$os = "linux"}}
+							{{- else if eq .OS "darwin" -}}
+							{{$os = "darwin"}}
+							{{- end -}}
```
Comment (Shikachuu) [inline pkg/get/tools.go:2588]: Correct me if I'm wrong here, but `bun` doesn't support windows, why should we handle it?

Also all the conditions returns the same as the `.OS` variable, why can't we just use it like `{{ $os := .OS }}` to reduce unnecessary complexity?
Or even better, can't we just use it like `{{ .OS }}` in the final value?

**Review by Shikachuu — CHANGES_REQUESTED** [review]

**Review by Shikachuu — APPROVED** [review]
LGTM!

**Comment by alexellis** [comment]
Thank you. Please could you let the Bun maintainer know or send him a Pr to the project docs to show arkade can install his tool?

**Comment by alexellis** [comment]
Thank you @Shikachuu, your contribution made it into a release: https://github.com/alexellis/arkade/releases/tag/0.8.33

**Pull request merged**